server/src/main/java/umm3601/user/UserController.java

```diff
@@ -113,24 +114,61 @@ public void getUsers(Context ctx) {
    * @param ctx a Javalin HTTP context
    */
   public void addNewUser(Context ctx) {
+    /*
+     * The follow chain of statements uses the Javalin validator system
+     * to verify that instance of `User` provided in this context is
+     * a "legal" user. It checks the following things (in order):
+     *    - The user has a value for the name (`usr.name != null`)
+     *    - The user name is not blank (`usr.name.length > 0`)
+     *    - The provided email is value (matches EMAIL_REGEX)
+     *    - The provided age is > 0
+     *    - The provided role is valid (one of "admin", "editor", or "viewer")
+     *    - A non-blank company is provided
+     */
     User newUser = ctx.bodyValidator(User.class)
-      .check(usr -> usr.name != null && usr.name.length() > 0) //Verify that the user has a name that is not blank
-      .check(usr -> usr.email.matches(emailRegex)) // Verify that the provided email is a valid email
-      .check(usr -> usr.age > 0) // Verify that the provided age is > 0
-      .check(usr -> usr.role.matches("^(admin|editor|viewer)$")) // Verify that the role is one of the valid roles
-      .check(usr -> usr.company != null && usr.company.length() > 0) // Verify that the user has a company that is not blank
+      .check(usr -> usr.name != null && usr.name.length() > 0)
+      .check(usr -> usr.email.matches(EMAIL_REGEX))
+      .check(usr -> usr.age > 0)
+      .check(usr -> usr.role.matches("^(admin|editor|viewer)$"))
+      .check(usr -> usr.company != null && usr.company.length() > 0)
       .get();
 
-    // Generate user avatar (you won't need this part for todos)
+    // Generate a user avatar (you won't need this part for todos)
+    newUser.avatar = generateAvatar(newUser.email);
+
+    userCollection.insertOne(newUser);
+    // 201 is the HTTP code for when we successfully
+    // create a new resource (a user in this case).
+    // See, e.g., https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
+    // for a description of the various response codes.
+    // The class HttpURLConnection contains named constants with
+    // most, but not all, of the standard HTTP status codes,
+    // including HTTP_CREATED for 201.
+    ctx.status(HttpURLConnection.HTTP_CREATED);
```

Comment:
I switched to using named constants for the HTTP response codes. We should probably try to do something like this in the JS/TS code, but I haven't done the hunting to find a library that defines all those named constants.

Reply:
Oh sweet, thanks! :)